javascript/javascript2/week1/index.js

```diff
@@ -0,0 +1,127 @@
+const recipeList = [
+  {
+    id: 1,
+    title: "Gløgg",
+    picture_url:
+      "https://upload.wikimedia.org/wikipedia/commons/thumb/e/e2/Gl%C3%B6gg_kastrull.JPG/800px-Gl%C3%B6gg_kastrull.JPG",
+    ingredients: [
+      { NAME: "Orange zest", AMOUNT: "0.5" },
+      { NAME: "Water", AMOUNT: "200 ml" },
+      { NAME: "Sugar", AMOUNT: "275 g" },
+      { NAME: "Whole cloves", AMOUNT: "5" },
+      { NAME: "Cinnamon sticks", AMOUNT: "2" },
+      { NAME: "Spice", AMOUNT: undefined },
+      { NAME: "Bottle of red wine", AMOUNT: "1" },
+      { NAME: "Raisins", AMOUNT: "100 g" },
+      { NAME: "Slipped Almonds", AMOUNT: "50 g" },
+    ],
+    description: "Mix everything, heat it, and you are good to go!",
+  },
+];
+
+let ingredientCount = 5;
+
+let insertButton = document.getElementById("button-insert");
+insertButton.addEventListener("click", function () {
+  let insertForm = document.getElementById("recipe-form");
+  insertForm.style.display = "block";
+});
+
+function renderRecipe() {
+  let secRecipe = document.getElementById("recipe");
+  secRecipe.replaceChildren();
+
+  recipeList.forEach((item) => {
+    let parent = document.createElement("div");
+    parent.classList.add("every-recipe");
+    parent.addEventListener("click", () => {
+      showModal(item);
+    });
+
+    let h3 = document.createElement("h3");
+    h3.innerHTML = item.title;
+    parent.appendChild(h3);
+
+    let img = document.createElement("img");
+    img.src = item.picture_url;
+    parent.appendChild(img);
+
+    secRecipe.appendChild(parent);
+  });
+}
+
+renderRecipe();
+
+let submitButton = document.getElementById("submit");
+submitButton.addEventListener("click", function (event) {
+  event.preventDefault();
+  let form = document.getElementById("recipe-form");
+  const data = Object.fromEntries(new FormData(form));
+  const recipe = convertFormToObject(data);
+  recipeList.push(recipe);
+  let insertForm = document.getElementById("recipe-form");
+  insertForm.style.display = "none";
+  renderRecipe();
+});
```

Comment:
You can add an` if` condition here to check if the user has filled in the required fields before adding a recipe. Otherwise, the user can submit an empty recipe and it will ruin your UI.